code/game/machinery/doors/airlock.dm

```diff
@@ -320,203 +359,112 @@ About the new airlock wires panel:
 /obj/machinery/door/airlock/proc/shock_user(mob/user, prob)
 	return (!issilicon(user) && isElectrified() && shock(user, prob))
 
-/obj/machinery/door/airlock/update_icon(state=0, override=0)
+/obj/machinery/door/airlock/update_icon(updates=ALL, state=0, override=FALSE)
```

Comment:
`state` should not be passed in here; and instead changed out of band.
I know i just mentioned that fixing this sort of thing is out of scope (and feel totally free to push back here!), but i think it's very worth normalizing the signatures of `update_icon` in this PR.

Here's a very easy workaround though: Simply create a new proc, and call the "proper" `update_icon` within:
```dm
/obj/machinery/door/airlock/proc/update_state(state=0, override=FALSE)
	if(operating && !override)
		return

	if(!state)
		state = density ? AIRLOCK_CLOSED : AIRLOCK_OPEN
	airlock_state = state

	update_icon(ALL)

	if(hasPower() && unres_sides)
		set_light(2, 1)
	else
		set_light(0)
```